packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol

```diff
@@ -72,6 +72,19 @@ contract PerpetualLiquidatable is PerpetualPositionManager {
         FixedPoint.Unsigned disputerDisputeRewardPct;
     }
 
+    // This struct is used in the `withdrawLiquidation` method that disperses liquidation and dispute rewards.
+    // `payToX` stores the total collateral to withdraw from the contract to pay X. This value might differ
+    // from `paidToX` due to precision loss between accounting for the `rawCollateral` versus the
+    // fee-adjusted collateral. These variables are stored within a struct to avoid the stack too deep error.
+    struct RewardsData {
```

Comment:
Would it make sense to add the `RewardsData` within the `LiquidationData` Struct? this way it will be persistent after the fact and can be seen after the liquidation is complete?

Reply:
In fact that doesn't work nor make sense as we delete the liquidation data once the withdrawal is finished.

Reply:
Yeah this struct is only initialized and modified within the `withdrawLiquidation` function. I originally wanted to just use `payToX` and `paidToX` as temporary variables within the function but due to scoping errors I decided to smash these into one struct.